src/player.cpp

```diff
@@ -10050,74 +10041,47 @@ void player::consume_effects(item *eaten, const it_comest *comest, bool rotten)
         // No good can come of this.
         return;
     }
-    if ( !(has_trait("GIZZARD")) && (rotten) && !(has_trait("SAPROPHAGE")) ) {
-        hunger -= rng(0, nutrition_for(comest));
-        thirst -= comest->quench;
+    float factor = 1.0;
+    float hunger_factor = 1.0;
+    bool unhealthy_allowed = true;
+
+    if (has_trait("GIZZARD")) {
+        factor *= .6;
+    }
+    if (has_trait("CARNIVORE")) {
+        // At least partially edible
+        if(eaten->made_of("flesh") || eaten->made_of("hflesh") || eaten->made_of("iflesh") ||
```

Comment:
Not really part of this PR, but for future references: `item` has nice function to check this, they can be used with (for example) static const vectors: `item::made_of_any` and `item::only_made_of`.  Using those with a static vector of material types is a bit easier to extend.